src/utilities.ts

```diff
@@ -43,6 +44,50 @@ export function getMinorFullName(acronym: string): string {
   return requirementJSON.minor[acronym].name;
 }
 
+export function getAllSubjects(): ReadonlySet<string> {
+  const set = new Set<string>();
+  fullCoursesArray.forEach(it => set.add(it.subject));
+  return set;
```

Comment:
Functionally, `return new Set<string>(fullCoursesArray.map(it => it.subject));`? Or something like that

Reply:
Mostly trying to save some space by avoiding constructing array with a lot of duplicate elements.